cloud_functions/climateiq_merge_scenario_predictions_cf/main.py

```diff
@@ -21,35 +22,89 @@
     r"(?P<batch_id>\w+)/(?P<prediction_type>\w+)/(?P<model_id>\w+)/"
     r"(?P<study_area_name>\w+)/(?P<scenario_id>\w+)/(?P<chunk_id>\w+)\.csv"
 )
+# ID for the Study Areas collection in Firestore.
+STUDY_AREAS_COLLECTION_ID = "study_areas"
+# ID for the Model collection in Firestore.
+MODEL_COLLECTION_ID = "Model"
+# ID for the Runs sub-collection in Firestore.
+RUNS_COLLECTION_ID = "runs"
 
 
-@functions_framework.http
-def merge_scenario_predictions(request: flask.Request) -> tuple[str, int]:
+@functions_framework.cloud_event
+def merge_scenario_predictions(cloud_event: http.CloudEvent):
     """Merges predictions for each chunk across scenarios into single files per chunk.
 
-    TODO: Trigger based on file writes instead.
+    Triggered by writes to the input bucket. If the input bucket finally contains all
+    the chunks and scenarios (which are listed in Firestore), then the merge is
+    performed.
 
     Args:
-        request: A Flask request with the query parameters: batch_id,
-            prediction_type, model_id, study_area_name.
-    Returns:
-        A tuple of the HTTP response (message, status_code).
+        cloud_event: The CloudEvent representing the storage event.
     """
+    data = cloud_event.data
+    object_name = data["name"]
+    match = re.match(CHUNK_FILE_NAME_PATTERN, object_name)
+    # Ignore files that don't match the pattern.
+    if not match:
+        return
+
+    batch_id, prediction_type, model_id, study_area_name = (
+        match.group("batch_id"),
+        match.group("prediction_type"),
+        match.group("model_id"),
+        match.group("study_area_name"),
+    )
+
```

Comment:
nit: use single try except block instead of 2 different ones? also add comment for why we print the error instead of raising an exception